packages/core/src/client.ts

```diff
@@ -1658,32 +1663,50 @@ export class MedplumClient extends EventTarget {
    * If the schema is already cached, the promise is resolved immediately.
    * @category Schema
    * @param profileUrl - The FHIR URL of the profile
-   * @returns Promise to a schema with the requested profile.
+   * @param options - (optional) Additional options
+   * @returns Promise with an array of URLs of the profile(s) loaded.
    */
-  requestProfileSchema(profileUrl: string): Promise<void> {
-    if (isProfileLoaded(profileUrl)) {
-      return Promise.resolve();
+  requestProfileSchema(profileUrl: string, options?: RequestProfileSchemaOptions): Promise<string[]> {
+    if (!options?.expandProfile && isProfileLoaded(profileUrl)) {
+      return Promise.resolve([profileUrl]);
     }
 
-    const cacheKey = profileUrl + '-requestSchema';
+    const cacheKey = profileUrl + '-requestSchema' + (options?.expandProfile ? '-nested' : '');
     const cached = this.getCacheEntry(cacheKey, undefined);
     if (cached) {
       return cached.value;
     }
 
-    const promise = new ReadablePromise<void>(
+    const promise = new ReadablePromise<string[]>(
       (async () => {
-        // Just sort by lastUpdated. Ideally, it would also be based on a logical sort of version
-        // See https://hl7.org/fhir/references.html#canonical-matching for more discussion
-        const sd = await this.searchOne('StructureDefinition', {
-          url: profileUrl,
-          _sort: '-_lastUpdated',
-        });
-
-        if (!sd) {
-          console.warn(`No StructureDefinition found for ${profileUrl}!`);
+        if (options?.expandProfile) {
+          const url = this.fhirUrl('StructureDefinition', '$expand-profile');
+          url.search = new URLSearchParams({ url: profileUrl }).toString();
+          const sds = await new ReadablePromise(
```

Comment:
I think this could be simplified (maybe)

If you omit `then(bundleToResourceArray)`, then `this.get()` will return a `Bundle<StructureDefinition>`

Then you could also omit `new ReadablePromise()`

Then you could replace the `for (const sd of sds)` with a call to `indexStructureDefinitionBundle()` (which basically just calls `loadDataType()` in a for loop.

The one thing you would still need to do is get a list of the URLs, but that might be simple enough.

```ts
return sdBundle.entry?.map((e) => e.resource.url);
```

Reply:
That would result in all the StructureDefinitions being loaded within the top-level profile's URL namespace which isn't how it works right now; each extension has `loadDataType` called with its own url. That decision had been made a while back without enough consideration...

I'd need to think through it a bit more, but I believe we could  actually make the change to have all extensions loaded/index behind the top-level profile's URL. Either way, I'd like to hold off on that change from this PR since it'd require touching some other code.

Reply:
Copy, sounds good.  Thanks